Improve basic usage (#382)
TalonCommunity/Wiki: docs/Basic Usage/basic_usage.md

Reviewer: I would put the customization at the very end. Also `customize websites` doesn't mean much without 
explaining the `go to <website>` command.

Author: Some of this is preexisting stuff I was not going to change in the first PR. I do agree that `go to <website>` is worth explaining.

> 

Reviewer: To me, pressing keys is the most basic functionality, I would explain that first before venturing on into dictation.

| `kebab`           | dashes instead of spaces                |
| `help formatters` | show all available formatters           |

### Dictating Individual Characters

#### Dictating Characters in Command Mode

`help symbols` shows the commands for dictating symbols. Numbers can be dictated in command or dictation mode by saying `numb <number>`, i.e. `numb 2000` to dictate `2000`. Individual letters can be dictated with the alphabet commands, which can be seen by saying `help alphabet`. A single capital letter can be dictated by saying `shift <user.letter>`, such as saying `shift air` to dictate `A`. Multiple capital letters can be dictated in command or dictation mode by saying `ship` before the letters, such as saying `ship air bat cap` to dictate `ABC`.

#### Dictating Characters in Dictation Mode

> In dication mode, lower case letters can be dictated by saying `spell` and then the letters, such as `spell air bat cap` to dictate `abc`. Keys can be pressed in dictation mode by saying `press` and then the key stroke. This can allow dictating symbols, such as `press tilde` to dictate `~`. `help punctuation` shows how to dictate punctuation symbols in dictation mode without the press prefix.

Reviewer 2: ```suggestion
In dictation mode, lowercase letters can be dictated by saying `spell` and then the letters, such as `spell air bat cap` to dictate `abc`. Keys can be pressed in dictation mode by saying `press` and then the key stroke. This can allow dictating symbols, such as `press tilde` to dictate `~`. `help punctuation` shows how to dictate punctuation symbols in dictation mode without the press prefix.
```

| Command                                   | Description                                               |
| ----------------------------------------- | --------------------------------------------------------- |
| `cap`                                     | capitalize the next word                                  |
| `no cap` or `no caps`                     | make the next word lowercase                              |
| `no space`                                | do not put a space before the next word                   |
| `cap that`                                | make the first word of the last dictated text capitalized |
| `no cap that` or `no caps that`           | make the first word of the last dictated text lowercase   |
| `formatted <user.formatters> <user.text>` | apply formatter to the dictated text                      |

The formatted command works with the formatters documented above. To give an example, saying `formatted title this is dictation` inserts the text `This Is Dictation`.

## Pressing Keys

Saying `help alphabet` shows the command for each letter key.

Saying `help symbols` shows the command for each symbol key.

> Keys can be combined with modifier keys, i.e saying `shift up` presses shift and up. Saying `help modifiers` shows the word to refer to each modifier key. The `super` key refers to the operating system specific modifier. This is used for the `windows` key on Windows and offers one way to refer to the `command` key on MacOS.

Reviewer 2: ```suggestion
Keys can be combined with modifier keys, i.e., saying `shift up` presses Shift and the up arrow key. Saying `help modifiers` shows the word to refer to each modifier key. The `super` key refers to the operating system specific modifier. This is used for the Windows logo key on Windows and offers one way to refer to the Command key on MacOS.
```

In a keystroke, number keys can be dictated without prefix. For instance, saying `control-1` presses the control and 1 keys at the same time.

> Keys can be pressed in dictation mode by saying `press` followed by the key stroke, such as saying `press shift up` to press shift and up. The `press` command can also be used in command or dictation mode to press modifier keys by themselves, such as `press control` to press the control key.

Reviewer 2: ```suggestion
Keys can be pressed in dictation mode by saying `press` followed by the key stroke, such as saying `press shift up` to press Shift and the up arrow key. The `press` command can also be used in command or dictation mode to press modifier keys by themselves, such as `press control` to press the Control key.
```

| `cap that`                                | make the first word of the last dictated text capitalized |
| `no cap that` or `no caps that`           | make the first word of the last dictated text lowercase   |
| `formatted <user.formatters> <user.text>` | apply formatter to the dictated text                      |

The formatted command works with the formatters documented above. To give an example, saying `formatted title this is dictation` inserts the text "This Is Dictation".

To improve automatic spacing and capitalization in dictation mode, you can enable the user.context_sensitive_dictation setting. You can enable this globally in the settings.talon file, which you can open by saying `customize settings`. Context sensitive dictation may be slow and may not work in some applications because it relies on selecting and copying the text near the cursor, so you may want to enable this on a per-application basis. See [this page on .talon files](Customization/talon-files.md) if you want to understand how to modify settings on a per context basis and create custom commands.

## Pressing Keys

Saying `help alphabet` shows the command for each letter key.

> Saying `help symbols` shows the command for each symbol key.

Reviewer 2: ```suggestion
In command mode, saying `help alphabet` shows spoken forms for letter keys. Saying `help symbols` shows spoken forms for symbol keys.
```

Keys can be combined with modifier keys, i.e., saying `shift up` presses <kbd>Shift</kbd> and the up arrow key <kbd>⇧</kbd>. Saying `help modifiers` shows the word to refer to each modifier key. The `super` key refers to the operating system specific modifier. This is used for the Windows logo key <kbd>⌘</kbd> on Windows and offers one way to refer to the<kbd>Command</kbd> key on MacOS.

In a keystroke, number keys can be dictated without prefix. For instance, saying `control one` presses the <kbd>control</kbd> and <kbd>1</kbd> keys at the same time.

> Keys can be pressed in dictation mode by saying `press` followed by the key stroke, such as saying `press shift up` to press <kbd>Shift</kbd> and the up arrow key <kbd>⇧</kbd>. The `press` command can also be used in command or dictation mode to press modifier keys by themselves, such as `press control` to press the <kbd>Control</kbd> key.

Reviewer 2: ```suggestion
Keys can be pressed in dictation mode by saying `press` followed by the keystroke, such as saying `press shift up` to press <kbd>Shift</kbd> and the up arrow key <kbd>↑</kbd>. The `press` command can also be used in command or dictation mode to press modifier keys by themselves, such as `press control` to press the <kbd>Control</kbd> key.
```

| `kebab`           | dashes instead of spaces                |
| `help formatters` | show all available formatters           |

> `help formatters` shows 2 types of formatters: prose formatters and code formatters. Code formatters are primarily for formatting identifiers and values in computer code. Prose formatters are primarily for ordinary text and are shown with a \* after them. When dictating text with a prose formatter, you can say the following in the middle of dictation.

Author: What behavioral differences did I leave out?

> `help formatters` shows 2 types of formatters: prose formatters and code formatters. Code formatters are primarily for formatting identifiers and values in computer code. Prose formatters are primarily for ordinary text and are shown with a \* after them. When dictating text with a prose formatter, you can say the following in the middle of dictation.

Reviewer 2: Punctuation :) if you can provide a quick note that it supports lots of post formatting things and we can add a link to a specific description of the prose capture in a different PR.

## Pressing Keys

In command mode, saying `help alphabet` shows spoken forms for letter keys. Saying `help symbols` shows spoken forms for symbol keys.

Keys can be combined with modifier keys, i.e., saying `shift up` presses <kbd>Shift</kbd> and the up arrow key <kbd>↑</kbd>. Saying `help modifiers` shows the word to refer to each modifier key. The `super` key refers to the operating system-specific modifier. This is used for the Windows logo key on Windows and offers one way to refer to the <kbd>Command</kbd> key on MacOS.

In a keystroke, number keys can be dictated without prefix. For instance, saying `control one` presses the <kbd>Control</kbd> and <kbd>1</kbd> keys at the same time.

Keys can be pressed in dictation mode by saying `press` followed by the keystroke, such as saying `press shift up` to press <kbd>Shift</kbd> and the up arrow key <kbd>↑</kbd>. The `press` command can also be used in command or dictation mode to press modifier keys by themselves, such as `press control` to press the <kbd>Control</kbd> key.

> Saying `help arrows` shows the names for referring to the arrow keys, which are just the direction names up, down, left, right. These can be used with a modifier or the `press` command but not by themselves.

Reviewer 2: ```suggestion
Saying `help arrows` shows spoken forms for the arrow keys, which are just the direction names up, down, left, right. These can be used with a modifier or the `press` command but not by themselves.
```

Saying `help special keys` shows the commands for other keys you can press, such as <kbd>Page Up</kbd> and <kbd>Escape</kbd>.

Function keys, like <kbd>f1</kbd>, can be dictated by saying `f` and then the key number.

> ### Dictation Mode Editing and Navigation

Reviewer: Why is this a subsection of "pressing keys"? Shouldnt it be in its own "editing commands" section?

| `select right <number_small> (word or words)`           | select the specified number of words to the right                                    |
| `select left <number_small> (character or characters)`  | select the specified number of characters to the left                                |
| `select right <number_small> (character or characters)` | select the specified number of characters to the right                               |
| `clear left <number_small> (word or words)`             | delete the specified number of words to the left                                     |
| `clear right <number_small> (word or words)`            | delete the specified number of words to the right                                    |
| `clear left <number_small> (character or characters)`   | delete the specified number of characters to the left                                |
| `clear right <number_small> (character or characters)`  | delete the specified number of characters to the right                               |

For example, saying `go up 10 lines` moves the cursor up 10 lines.

### Command Mode Editing and Navigation

#### Edit/Navigation Actions and Modifiers

> Command mode provides flexible editing and navigation commands where an editing/navigation command can be followed by a modifier to apply it to.

Reviewer 2: ```suggestion
Command mode provides flexible editing and navigation commands.
```

| Command                   | Description                                                                       |
| ------------------------- | --------------------------------------------------------------------------------- |
| `mouse grid`              | open the mouse grid                                                               |
| `grid win`                | open the mouse grid on the window                                                 |
| `grid screen <number>`    | open the mouse grid on the specified screen                                       |
| `grid <user.number_key>+` | open the mouse grid and narrow it using the numbers from the dictated number keys |
| `grid close`              | close the mouse grid                                                              |

### Cursor Hide/Show Commands

| Command     | Description                             |
| ----------- | --------------------------------------- |
| `curse yes` | hides the mouse cursor for eye tracking |
| `curse no`  | shows the mouse cursor                  |

These commands require activating the `user.mouse_cursor_commands_enable` tag (see Community settings.talon).

> ## Working With Media

Reviewer: ```suggestion
## Media and Volume Control
```
